### PR TITLE
patch: update to latest core 0.6.0 & base 0.6.1

### PR DESCRIPTION
**Pull request by OveBDev**

### CODE REVIEW

File: src/main.ts [hunk 1]
```diff
@@ -282,9 +281,9 @@ async function toggleMockDataClicked() {
 }
 
 async function testCommReturnTypesClicked(): Promise<void> {
-    const result = (await echoSearchWorker.testCommReturnTypes()) as ErrorForTesting;
+    const result = await echoSearchWorker.testCommReturnTypes(); //as ErrorForTesting;
     log.info('in main', result);
-    console.log({ ...result });
+    //console.log({ ...result });
```
Comment (CsabaTheKiss) [inline src/main.ts:286]: commented line, is it still needed?

File: src/results/convertToError.ts [hunk 2]
```diff
@@ -1,6 +1,13 @@
 import { BaseError } from '@equinor/echo-base';
 
-export class NotAProperErrorObject extends BaseError {}
+export class NotAProperErrorObject extends BaseError {
+    /**
+     *
```
Comment (CsabaTheKiss) [inline src/results/convertToError.ts:5]: Missing doc? 🤔

File: src/results/errorConversion.test.ts [hunk 3]
```diff
@@ -85,16 +98,22 @@ describe('errorFromException() back to toBaseError() should be equal', () => {
...
         const actual = toBaseError(resultWithError.error ?? { type: SyncErrorType.Unknown });
 
         //then
-        expect(actual).toEqual(baseError);
+        //comparing Error/CustomError is not reliable, we have to compare all of these:
+        expect(actual.name).toEqual(baseError.name);
+        expect(actual.message).toEqual(baseError.message);
+        expect(actual.stack).toEqual(baseError.stack);
+        expect(actual).toBeInstanceOf(ForbiddenError);
+        expect(JSON.stringify(actual)).toStrictEqual(JSON.stringify(baseError));
```
Comment (CsabaTheKiss) [inline src/results/errorConversion.test.ts:113]: Minor comment: Yep, this is also a way to do it. Isn't just this enough? As I know, this takes all properties, so maybe there's no need for the checks in line 109-111.

File: src/results/errorConversion.ts [hunk 4]
```diff
@@ -80,22 +81,40 @@ export function createResultErrorFromException(errorArg: Error | BaseError | unk
...
+        httpStatusCode: findPropertyByName(error, 'httpStatusCode') as number,
+        url: findPropertyByName(error, 'url') as string,
+        extraBaseProperties: extraBaseProperties,
+        innerProperties: innerProperties
     };
 
     return createError(searchModuleError);
 }
+
+function stripCommonPropertiesFromBaseError(properties?: Record<string, unknown>): Record<string, unknown> | undefined {
```
Comment (CsabaTheKiss) [inline src/results/errorConversion.ts:107]: Why is this needed? : O (just out of curiosity)

File: src/results/errorConversion.ts [hunk 5]
```diff
@@ -80,22 +81,40 @@ export function createResultErrorFromException(errorArg: Error | BaseError | unk
...
+    if (!properties) return undefined;
+
+    const props = { ...properties };
+
+    const propertyNamesToDelete = ['name', 'message', 'stack', 'errorTraceId', 'innerError'];
+    propertyNamesToDelete.forEach((propertyName) => {
+        if ((props && props[propertyName]) || props[propertyName] === undefined) {
```
Comment (CsabaTheKiss) [inline src/results/errorConversion.ts:114]: Hmmm, at first glance this logical expression will be always true (except when props[propertyName] = null).

If this above one is true, then why do we need to delete the prop in question? It's already undefined (not set) 🤔 
`props[propertyName] === undefined`

File: src/workers/externalCalls.ts [hunk 6]
```diff
@@ -279,6 +279,7 @@ export function externalTestCommReturnTypes(): ErrorForTesting {
     });
 
     console.log("error'en", { ...err });
+    //return err;
```
Comment (CsabaTheKiss) [inline src/workers/externalCalls.ts:282]: Commented out code here. Is it still needed?

File: src/results/errorConversion.ts (hunk 4 above)
Comment (OveBDev) [reply src/results/errorConversion.ts:107]: The problem is that commlink will not preserve custom properties of an error-class between main thread and worker thread. (maybe same reason unit tests of base error was not checking all properties)
So I have these helper functions for converting a BaseError (worker) => interface => back to BaseError (main)
Only the first level of Error is converted, and innerProperties are just kept as a record on the main thread instead.

This function removes baseError properties which are handled separately. Any additional custom properties we want to keep are kept as extraBaseProperties, and added as innerError. This unit test is an example:
should convert 'CustomTestError extending Error' to interface SearchModuleError, and back to baseError with preserved fields and custom properties as innerError

File: src/results/errorConversion.ts (hunk 5 above)
Comment (OveBDev) [reply src/results/errorConversion.ts]: Agree, just me becoming blind sometimes :) no need to overcomplicate stuff, refactored now.

File: src/results/errorConversion.test.ts (hunk 3 above)
Comment (OveBDev) [reply src/results/errorConversion.test.ts:113]: image
This is where error behaves strange, as you can see even JSON is not able to get all properties, like message and stack. And I think it's better to just keep all (including name), so that we easily can see that this has been tested in the future, without having to wonder about how error stringify works.

File: src/results/errorConversion.ts (hunk 4 above)
Comment (CsabaTheKiss) [reply src/results/errorConversion.ts]: Interesting, thanks for the explanation!

File: src/results/errorConversion.test.ts (hunk 3 above)
Comment (CsabaTheKiss) [reply src/results/errorConversion.test.ts:113]: Ah, I see! 👍

**Review by CsabaTheKiss — APPROVED** [review]
